Fix runtime error when running with WEKA-3.8.5

justinuliu opened the pull request:
Hi,

I have fixed the error about "Error marshalling XML response" when running Auto-WEKA with WEKA-3.8.5. The problem is that java11 has removed JAXB module which is used by Auto-WEKA.

I find that WEKA-3.8.5 has already included the necessary packages (jakarta.bine-api:2.3.3 and jaxb-runtime:2.3.3 etc.) but Auto-WEKA do not have the binding information of the implementation classes at runtime. A simple solution is to add the binding information (in jaxb.properties) with the class XmlSerializable.

Could you please consider merging this PR and release a new version.

Thank you,
Justin Liu

Commit:
Fix runtime error when running with WEKA-3.8.5

The problem is that java11 has removed JAXB module which is used by Auto-WEKA.

I find that WEKA-3.8.5 has already included the necessary packages (jakarta.bine-api:2.3.3 and jaxb-runtime:2.3.3 etc.) but Auto-WEKA do not have the binding information of the implementation classes at runtime. A simple solution is to add the binding information (in jaxb.properties) with the class XmlSerializable.

justinuliu commented:
> Thanks -- it looks like Travis is broken though and the automatic testing wasn't run. Did you check that all tests pass?

Yes, 14 tests are passed with 1 ignored of total 15.

larskotthoff commented:
Hmm, I'm afraid I'm getting multiple failures:

```
test:
    [junit] Testsuite: autoweka.AutoWEKATestSuite
    [junit] .\
    [junit] bar=foo:prop=value=something__COLONESCAPE__:else=foo
    [junit] Result value=something:else=foo
    [junit] This,
    [junit] is,
    [junit] my split,
    [junit] string" "with,
    [junit] lotsC:\foath\fishof\,
    [junit] o',
    [junit] fish,
    [junit] "and,
    [junit] even,
    [junit] escaped",
    [junit] 2021-06-04 10:43:26.199 [main] ERROR XmlSerializable - Error unmarshalling XML response
    [junit] caught a FileNotFoundException!
    [junit] Tests run: 14, Failures: 0, Errors: 2, Skipped: 1, Time elapsed: 1.559 sec
    [junit] ------------- Standard Output ---------------
    [junit] .\
    [junit] bar=foo:prop=value=something__COLONESCAPE__:else=foo
    [junit] Result value=something:else=foo
    [junit] This,
    [junit] is,
    [junit] my split,
    [junit] string" "with,
    [junit] lotsC:\foath\fishof\,
    [junit] o',
    [junit] java.lang.RuntimeException: Error unmarshalling XML response
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:81)
    [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
    [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
    [junit] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    [junit] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    [junit] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    [junit] fish,
    [junit] "and,
    [junit] even,
    [junit] escaped",
    [junit] caught a FileNotFoundException!
    [junit] ------------- ---------------- ---------------
    [junit] ------------- Standard Error -----------------
    [junit] 2021-06-04 10:43:26.199 [main] ERROR XmlSerializable - Error unmarshalling XML response
    [junit] java.lang.RuntimeException: Error unmarshalling XML response
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:81)
    [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
    [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
    [junit] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
    [junit] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    [junit] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
    [junit] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    [junit] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
    [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
    [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
    [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    [junit] 	at org.junit.runners.Suite.runChild(Suite.java:127)
    [junit] 	at org.junit.runners.Suite.runChild(Suite.java:26)
    [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    [junit] 	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:535)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1197)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1042)
    [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
    [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
    [junit] 	... 32 more
    [junit] ------------- ---------------- ---------------
    [junit] 
    [junit] Testcase: expandPathTest took 0.04 sec
    [junit] Testcase: playpen took 0.001 sec
    [junit] Testcase: splitTest took 0.001 sec
    [junit] Testcase: levelSubsets took 0 sec
    [junit] 	SKIPPED
    [junit] Testcase: levelSizes took 1.45 sec
    [junit] Testcase: argumentConversionTest1 took 0.003 sec
    [junit] Testcase: argumentConversionTest2 took 0 sec
    [junit] Testcase: argumentConversionTest3 took 0.001 sec
    [junit] Testcase: argumentConversionTest4 took 0.001 sec
    [junit] Testcase: settersAndAveragerTest took 0.004 sec
    [junit] Testcase: compareToTest took 0 sec
    [junit] Testcase: mergeTest took 0.001 sec
    [junit] Testcase: rankerTest took 0.032 sec
    [junit] 	Caused an ERROR
    [junit] Failed to load from XML file
    [junit] java.lang.RuntimeException: Failed to load from XML file
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:86)
    [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
    [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
    [junit] 
    [junit] Testcase: parseTrajectoryTest took 0.003 sec
    [junit] 	Caused an ERROR
    [junit] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    [junit] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
    [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
    [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
    [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    [junit] 	at org.junit.runners.Suite.runChild(Suite.java:127)
    [junit] 	at org.junit.runners.Suite.runChild(Suite.java:26)
    [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
    [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
    [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
    [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
    [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
    [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
    [junit] 	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:535)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1197)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1042)
    [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
    [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
    [junit] 	... 32 more
    [junit] java.lang.RuntimeException: Error unmarshalling XML response
    [junit] java.lang.RuntimeException: java.lang.RuntimeException: Error unmarshalling XML response
    [junit] 	at autoweka.Experiment.createFromFolder(Experiment.java:317)
    [junit] 	at autoweka.smac.SMACTrajectoryParserTester.parseTrajectoryTest(SMACTrajectoryParserTester.java:22)
    [junit] Caused by: java.lang.RuntimeException: Error unmarshalling XML response
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
    [junit] 	at autoweka.Experiment.fromXML(Experiment.java:183)
    [junit] 	at autoweka.Experiment.createFromFolder(Experiment.java:315)
    [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
    [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
    [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
    [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
    [junit] 
```

justinuliu commented:
> Hmm, I'm afraid I'm getting multiple failures:
> 
> ```
> test:
>     [junit] Testsuite: autoweka.AutoWEKATestSuite
>     [junit] .\
>     [junit] bar=foo:prop=value=something__COLONESCAPE__:else=foo
>     [junit] Result value=something:else=foo
>     [junit] This,
>     [junit] is,
>     [junit] my split,
>     [junit] string" "with,
>     [junit] lotsC:\foath\fishof\,
>     [junit] o',
>     [junit] fish,
>     [junit] "and,
>     [junit] even,
>     [junit] escaped",
>     [junit] 2021-06-04 10:43:26.199 [main] ERROR XmlSerializable - Error unmarshalling XML response
>     [junit] caught a FileNotFoundException!
>     [junit] Tests run: 14, Failures: 0, Errors: 2, Skipped: 1, Time elapsed: 1.559 sec
>     [junit] ------------- Standard Output ---------------
>     [junit] .\
>     [junit] bar=foo:prop=value=something__COLONESCAPE__:else=foo
>     [junit] Result value=something:else=foo
>     [junit] This,
>     [junit] is,
>     [junit] my split,
>     [junit] string" "with,
>     [junit] lotsC:\foath\fishof\,
>     [junit] o',
>     [junit] java.lang.RuntimeException: Error unmarshalling XML response
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:81)
>     [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
>     [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
>     [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
>     [junit] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
>     [junit] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
>     [junit] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
>     [junit] fish,
>     [junit] "and,
>     [junit] even,
>     [junit] escaped",
>     [junit] caught a FileNotFoundException!
>     [junit] ------------- ---------------- ---------------
>     [junit] ------------- Standard Error -----------------
>     [junit] 2021-06-04 10:43:26.199 [main] ERROR XmlSerializable - Error unmarshalling XML response
>     [junit] java.lang.RuntimeException: Error unmarshalling XML response
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:81)
>     [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
>     [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
>     [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
>     [junit] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
>     [junit] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
>     [junit] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
>     [junit] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
>     [junit] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
>     [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
>     [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
>     [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
>     [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
>     [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
>     [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
>     [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
>     [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
>     [junit] 	at org.junit.runners.Suite.runChild(Suite.java:127)
>     [junit] 	at org.junit.runners.Suite.runChild(Suite.java:26)
>     [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
>     [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
>     [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
>     [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
>     [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
>     [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
>     [junit] 	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:535)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1197)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1042)
>     [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
>     [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
>     [junit] 	... 32 more
>     [junit] ------------- ---------------- ---------------
>     [junit] 
>     [junit] Testcase: expandPathTest took 0.04 sec
>     [junit] Testcase: playpen took 0.001 sec
>     [junit] Testcase: splitTest took 0.001 sec
>     [junit] Testcase: levelSubsets took 0 sec
>     [junit] 	SKIPPED
>     [junit] Testcase: levelSizes took 1.45 sec
>     [junit] Testcase: argumentConversionTest1 took 0.003 sec
>     [junit] Testcase: argumentConversionTest2 took 0 sec
>     [junit] Testcase: argumentConversionTest3 took 0.001 sec
>     [junit] Testcase: argumentConversionTest4 took 0.001 sec
>     [junit] Testcase: settersAndAveragerTest took 0.004 sec
>     [junit] Testcase: compareToTest took 0 sec
>     [junit] Testcase: mergeTest took 0.001 sec
>     [junit] Testcase: rankerTest took 0.032 sec
>     [junit] 	Caused an ERROR
>     [junit] Failed to load from XML file
>     [junit] java.lang.RuntimeException: Failed to load from XML file
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:86)
>     [junit] 	at autoweka.ConfigurationCollection.fromXML(ConfigurationCollection.java:55)
>     [junit] 	at autoweka.ConfigurationTester.rankerTest(ConfigurationTester.java:151)
>     [junit] 
>     [junit] Testcase: parseTrajectoryTest took 0.003 sec
>     [junit] 	Caused an ERROR
>     [junit] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
>     [junit] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
>     [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
>     [junit] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
>     [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
>     [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
>     [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
>     [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
>     [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
>     [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
>     [junit] 	at org.junit.runners.Suite.runChild(Suite.java:127)
>     [junit] 	at org.junit.runners.Suite.runChild(Suite.java:26)
>     [junit] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
>     [junit] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
>     [junit] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
>     [junit] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
>     [junit] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
>     [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
>     [junit] 	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:535)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1197)
>     [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1042)
>     [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
>     [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
>     [junit] 	... 32 more
>     [junit] java.lang.RuntimeException: Error unmarshalling XML response
>     [junit] java.lang.RuntimeException: java.lang.RuntimeException: Error unmarshalling XML response
>     [junit] 	at autoweka.Experiment.createFromFolder(Experiment.java:317)
>     [junit] 	at autoweka.smac.SMACTrajectoryParserTester.parseTrajectoryTest(SMACTrajectoryParserTester.java:22)
>     [junit] Caused by: java.lang.RuntimeException: Error unmarshalling XML response
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:107)
>     [junit] 	at autoweka.Experiment.fromXML(Experiment.java:183)
>     [junit] 	at autoweka.Experiment.createFromFolder(Experiment.java:315)
>     [junit] Caused by: javax.xml.bind.JAXBException: jaxb.properties in package autoweka does not contain the javax.xml.bind.context.factory property.
>     [junit] 	at javax.xml.bind.ContextFinder.find(ContextFinder.java:392)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:641)
>     [junit] 	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:584)
>     [junit] 	at autoweka.XmlSerializable.fromXML(XmlSerializable.java:101)
>     [junit] 
> ```

Could you tell me which version of JAVA and WEKA do you use? I think it might be a backward compatibility problem.

larskotthoff commented:
```
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
```

And the WEKA version included with Auto-WEKA.

justinuliu committed:
Backward compatibility for JAXB

Following JAXB requirement, we must provide javax.xml.bind.context.factory property in autoweka for the backward compatibility when we use java 8,9 and 10.

justinuliu commented:
> ```
> openjdk version "1.8.0_292"
> OpenJDK Runtime Environment (build 1.8.0_292-b10)
> OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
> ```
> 
> And the WEKA version included with Auto-WEKA.

I have fixed this error by adding a commit. Please make another check with it. Thank you.

larskotthoff commented:
Thanks, looks like this works. I'll merge and make a new release.

celestinoxp commented:
I try use java 11 jdk FX and works fine. i think the magic is the FX of java
